lighthouse-core/lib/sentry.js

```diff
@@ -3,94 +3,95 @@
+  captureException: async () => {},
 };
 
-const SAMPLED_ERRORS = [
-  // Error code based sampling
-  {pattern: /DOCUMENT_REQUEST$/, rate: 0.1},
-  {pattern: /(IDLE_PERIOD|FMP_TOO_LATE)/, rate: 0.1},
-  {pattern: /^NO_.*/, rate: 0.1},
-  // Message based sampling
-  {pattern: /Could not load stylesheet/, rate: 0.01},
-  {pattern: /Failed to decode/, rate: 0.1},
-  {pattern: /All image optimizations failed/, rate: 0.1},
-  {pattern: /No.*resource with given/, rate: 0.01},
-  {pattern: /No.*node with given id/, rate: 0.01},
-];
-
 /**
- * We'll create a delegate for sentry so that environments without error reporting enabled will use
- * noop functions and environments with error reporting will call the actual Sentry methods.
+ * When called, replaces noops with actual Sentry implementation.
+ * @param {{url: string, flags: LH.CliFlags, environmentData: ConstructorOptions}} opts
  */
-const sentryDelegate = Object.assign({}, sentryApi);
-sentryDelegate.init = function init(opts) {
+function init(opts) {
+  // If error reporting is disabled, leave the functions as a noop
   if (!opts.flags.enableErrorReporting) {
-    // If error reporting is disabled, leave the functions as a noop
     return;
   }
 
-  const environmentData = opts.environmentData || {};
-  const sentryConfig = Object.assign({}, environmentData, {captureUnhandledRejections: true});
+  // If not selected for samping, leave the functions as a noop.
+  if (SAMPLE_RATE <= Math.random()) {
+    return;
+  }
 
   try {
     const Sentry = require('raven');
+    const sentryConfig = Object.assign({}, opts.environmentData,
+      {captureUnhandledRejections: true});
     Sentry.config(SENTRY_URL, sentryConfig).install();
-    Object.keys(sentryApi).forEach(functionName => {
-      // Have each delegate function call the corresponding sentry function by default
-      sentryDelegate[functionName] = (...args) => Sentry[functionName](...args);
-    });
 
-    // Special case captureException to return a Promise so we don't process.exit too early
-    sentryDelegate.captureException = (err, opts) => {
-      opts = opts || {};
+    // Have each delegate function call the corresponding sentry function by default
+    sentryDelegate.captureMessage = (...args) => Sentry.captureMessage(...args);
+    sentryDelegate.captureBreadcrumb = (...args) => Sentry.captureBreadcrumb(...args);
+    sentryDelegate.getContext = () => Sentry.getContext();
 
-      const empty = Promise.resolve();
+    // Special case captureException to return a Promise so we don't process.exit too early
+    sentryDelegate.captureException = async (err, opts = {}) => {
       // Ignore if there wasn't an error
-      if (!err) return empty;
+      if (!err) return;
+
       // Ignore expected errors
-      if (err.expected) return empty;
-      // Sample known errors that occur at a high frequency
-      const sampledErrorMatch = SAMPLED_ERRORS.find(sample => sample.pattern.test(err.message));
```

Comment:
can we keep this block?

along with something like this above:

```js
const SAMPLED_ERRORS = [	
  // Error code based sampling	
  // e.g.: {pattern: /No.*node with given id/, rate: 0.01},	
];
```

Reply:
There you go :P